pypeit/spectrographs/keck_lris.py

```diff
@@ -136,20 +136,40 @@ def init_meta(self):
         # Ingest
         self.meta = meta
 
+    # TODO Compound meta should have an ignore_bad_headers which should be True by default. Same goes for other meta data.
     def compound_meta(self, headarr, meta_key):
         if meta_key == 'binning':
             binspatial, binspec = parse.parse_binning(headarr[0]['BINNING'])
             binning = parse.binning2string(binspec, binspatial)
             return binning
         elif 'lampstat' in meta_key:
             idx = int(meta_key[-2:])
-            curr_date = time.Time(headarr[0]['MJD-OBS'], format='mjd')
+            # JFH: Since the golden rule is pypeit_setup should NEVER CRASH, even if headers are corrupt, we throw
+            # warnings and attempt to determine the date as best we can. If not a bogus MJD is used.
+            # TODO Which BOGUS MJD should we use??
+            try:
+                curr_date = time.Time(headarr[0]['MJD-OBS'], format='mjd')
+            except KeyError:
+                msgs.warn("This file does not have the MJD-OBS header keyword. "
+                          "Check your data headers, something is probably wrong. We will try to use 'DATE' instead")
+                try:
+                    curr_date = time.Time(headarr[0]['MJD-OBS'], format='mjd')
+                except KeyError:
+                    msgs.warn("Could not parse date from your header. Using 1/1/2020 instead to avoid a crash. Check your headers!!!")
+                    # JFH 58849 = 1/1/2020
+                    curr_date = time.Time(58849, format='mjd')
             # Modern -- Assuming the change occurred with the new red detector
             t_newlamp = time.Time("2014-02-15", format='isot')  # LAMPS changed in Header
             if curr_date > t_newlamp:
                 lamp_names = ['MERCURY', 'NEON', 'ARGON', 'CADMIUM', 'ZINC', 'KRYPTON', 'XENON',
                               'FEARGON', 'DEUTERI', 'FLAMP1', 'FLAMP2', 'HALOGEN']
-                return headarr[0][lamp_names[idx-1]]  # Use this index is offset by 1
+                # TODO Make this a sens
+                try:
```

Comment:
same as above

Reply:
So here the try except seems like the most compact solution. Otherwise I need to loop over all of those header cards, check whether they are in the headarr, and then return the ones that are. 

Are treatment of header cards in these meta data routines is a bit slipshod. It is constantly assuming certain header cards exist, which I have to then go in and fixed in various places when I encounter corrupt headers that cause crashes. We can assume and require that stuff created by PypeIt has the write headers and version (in some cases not as we discussed), but we should assume and require nothing of files from the telescope, and never crash for bad headers.

Reply:
I don't agree with "we should assume and require nothing of files from the telescope, and never crash for bad headers"
and indeed that is why it is coded this way

Reply:
Let's discuss at the meeting today, I strongly feel that pypeit_setup should never crash.

Reply:
i'll butt in and opine that crashing ungracefully on an exception should never be part of a software design and always be considered a bug. in the case of setup, if it runs into a problem with an image, there are three appropriate options as i see it:

1. skip the image entirely, spew a warning, and don't include it in the output. 
2. include it in the output, but flag it as bad or comment it out.
3. stop the setup script gracefully with a human readable error message.

which of these is most appropriate is situation-dependent so it may be useful to add flags to select which path to take. at least make the default behavior 3 so that if it does crash, it does so usefully with information as to why it did.

Reply:
I agree completely @tepickering. Currently the code requires headers exist, and crashes if they don't with a cryptic error message, and there are no options to skip the bad files or exit gracefully. 

If you are observing at the telescope and constantly updating your raw data directory, then every time you try to re-run pypeit_setup you will first need to move the corrupt file somewhere after every time you rsync the data over.